File: coding/14th May 2025/Longest Strictly Increasing or Strictly Decreasing Subarray.java
//Longest Strictly Increasing or Strictly Decreasing Subarray
class Solution 
{
    public int longestMonotonicSubarray(int[] nums) 
    {
        int res = 1;
        int ic = 1, dc = 1;
        for(int i = 1; i < nums.length; i++)
        {
            if(nums[i - 1] < nums[i])
            {
                ic++;
                dc = 1;
            }
            else if(nums[i - 1] > nums[i])
            {
                ic = 1;
                dc++;
            }
            else
            {
                ic = 1;
                dc = 1;
            }
            res = Math.max(res, Math.max(ic, dc));
        }
        return res;
    }
}

File: Coding in Python/17th January 2026/Longest Strictly Increasing or Strictly Decreasing Subarray.py
#Longest Strictly Increasing or Strictly Decreasing Subarray
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        res = count = 0
        prev = nums[0]
        op, other = operator.lt, operator.gt
        for i in nums:
            if op(i, prev):
                count += 1
            else:
                res = max(res, count)
                op, other = other, op
                count = 1 if i == prev else 2
            prev = i
        return max(res, count)
